format/Flight.proto

```diff
@@ -189,6 +191,31 @@ message Result {
   bytes body = 1;
 }
 
+/*
+ * The result of the CancelFlightInfo action.
+ *
+ * The result should be stored in Result.body.
+ */
+message ActionCancelFlightInfoResult {
+  enum CancelResult {
+    // The cancellation status is unknown. Servers should avoid using
+    // this value (send a NOT_FOUND error if the requested query is
+    // not known). Clients can retry the request.
+    CANCEL_RESULT_UNSPECIFIED = 0;
```

Comment:
`CANCEL_RESULT_` prefix may be redundant.

```suggestion
    UNSPECIFIED = 0;
```

Reply:
We should use prefix.

https://protobuf.dev/programming-guides/style/#enums

> Prefer prefixing enum values instead of surrounding them in an enclosing message.

Reply:
Yes, the reason (in case you didn't find it) is that enums in Protobuf use C-style namespacing, which is carried over into generated code. So it will end up as `ActionCancelFlightInfoResult::[CANCEL_RESULT_]UNSPECIFIED`, and so if you ever need a second enum, you need to prefix the name.

Reply:
@lidavidm ping

Reply:
> Can we use CancelStatus or something instead of CancelResult to use Result in CancelResult and ActionCancelFlightInfoResult?

Sorry, what are you proposing here?

> I moved CancelResult to top-level to reduce ActionCancelFlightInfoResult::... prefix.

I think that's fine.

Reply:
> Sorry, what are you proposing here?

I wanted to say that how about the following (or other better name)?

```diff
index cda7dae31..5ccdd5b9a 100644
--- a/format/Flight.proto
+++ b/format/Flight.proto
@@ -196,20 +196,20 @@ message Result {
  *
  * This is used by CancelFLightInfoResult.result.
  */
-enum CancelResult {
+enum CancelStatus {
   // The cancellation status is unknown. Servers should avoid using
   // this value (send a NOT_FOUND error if the requested query is
   // not known). Clients can retry the request.
-  CANCEL_RESULT_UNSPECIFIED = 0;
+  CANCEL_STATUS_UNSPECIFIED = 0;
   // The cancellation request is complete. Subsequent requests with
   // the same payload may return CANCELLED or a NOT_FOUND error.
-  CANCEL_RESULT_CANCELLED = 1;
+  CANCEL_STATUS_CANCELLED = 1;
   // The cancellation request is in progress. The client may retry
   // the cancellation request.
-  CANCEL_RESULT_CANCELLING = 2;
+  CANCEL_STATUS_CANCELLING = 2;
   // The query is not cancellable. The client should not retry the
   // cancellation request.
-  CANCEL_RESULT_NOT_CANCELLABLE = 3;
+  CANCEL_STATUS_NOT_CANCELLABLE = 3;
 }
 
 /*
@@ -218,7 +218,7 @@ enum CancelResult {
  * The result should be stored in Result.body.
  */
 message CancelFlightInfoResult {
-  CancelResult result = 1;
+  CancelStatus status = 1;
 }
 
 /*
```

Reply:
Ah, to avoid confusion with Arrow `Result`? 

I guess Arrow also uses `Status`, so both are the same to me. If using `CancelStatus` is clearer to you, then it sounds good to me.

Reply:
> Ah, to avoid confusion with Arrow `Result`?

No. I just want to avoid `Result` duplication in `CancelFlightInfoResult` and `CancelResult`.

Reply:
BTW, we have `message Result` in `Flight.proto`. It may confuse us with Arrow `Result`...

Reply:
> BTW, we have `message Result` in `Flight.proto`. It may confuse us with Arrow `Result`...

Yes, the Flight one actually happened to come first, but it's unfortunate...

> No. I just want to avoid Result duplication in CancelFlightInfoResult and CancelResult.

Ok. I think `status` is also good then.

Reply:
OK!